flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/split/MySqlRecords.java

```diff
@@ -65,7 +72,12 @@ public SourceRecord nextRecordFromSplit() {
         final Iterator<SourceRecord> recordsForSplit = this.recordsForCurrentSplit;
         if (recordsForSplit != null) {
             if (recordsForSplit.hasNext()) {
-                return recordsForSplit.next();
+                SourceRecord sourceRecord = recordsForSplit.next();
+                Long messageTimestamp = RecordUtils.getMessageTimestamp(sourceRecord);
+                if (messageTimestamp != null) {
+                    metricRecorder.setFetchDelay(fetchTimestamp - messageTimestamp);
+                }
```

Comment:
This is not the time point when the MySqlSplitReader fetched the records, because this will be called in `SourceReaderBase#pollNext` which is always later than the time we fetched it.

Reply:
we can use `ts_ms` field in SourceRecord  as the `fetchTimestamp` which represents the time that data enter debezium, it also means the time data arrive Source Operator